Improve Windows installer documentation (#9753)
DataDog/documentation: content/en/agent/faq/windows-agent-ddagent-user.md

kind: faq
---

Starting with release `6.11.0`, the Core and APM/Trace components of the Windows Agent run under the a dedicated user account, instead of running under the `LOCAL_SYSTEM` account, as was the case on prior versions. If enabled, the Live Process component still runs under the `LOCAL_SYSTEM` account.

> The Agent can either use a user-supplied account or create one.

Reviewer: ```suggestion
The Agent installer creates a new account by default (`ddagentuser`) but it can also use a user-supplied account.
```
Let's mention the default early, since it should Just Work (TM) for most users and they can stop reading here.

* It has remote login disabled
* It has network login disabled

**Note**: Since the user is modified during installation to restrict its privileges, including login privileges, make sure it is not a 'real' user account but an account solely dedicated to run the Datadog Agent.

## Installation

If no user is specified on the command line, the installer will attempt to create a local user named `ddagentuser` with a randomly generated password.

If a user is specified on the command line, but this user is not found on the system, the installer will attempt to create it.
If a password was specified, the installer will use that password, otherwise it will generate a random password.

To specify a username and/or password on the command line pass the following properties to the msiexec:

```shell
Msiexec /i ddagent.msi DDAGENTUSER_NAME=<USERNAME> DDAGENTUSER_PASSWORD=<PASSWORD>
```

**Note**: The `<USERNAME>` must be 20 characters or less in order to comply with Microsoft's [Active Directory Schema (AD Schema) SAM-Account-Name attribute][1].

> **Note**: Due to a restriction in the MSI installer, the `DDAGENTUSER_PASSWORD` property cannot contain the semicolon character ';'.

Reviewer: No longer true after your changes. Should we still mention this?

It must be separated from the `<USERNAME>` with a `\`.

**Note**: The `<USERNAME>` must be 20 characters or less in order to comply with Microsoft's [Active Directory Schema (AD Schema) SAM-Account-Name attribute][1].

**Note**: Due to a restriction in the MSI installer, the `DDAGENTUSER_PASSWORD` property cannot contain the semicolon character ';'.

#### Domain controllers

##### Primary and backup Domain Controllers

When installing the Agent on a Domain Controller, there is no notion of 'local user', therefore if the installer creates a user it will be a domain user rather than a local one.

If no user is specified on the command line, the installer will create a *Domain* account named 'ddagentuser' in the controller's domain.

If a user is specified on the command line, but this user is not found in the Domain, the installer will attempt to create it. If a password was specified, the installer will use that password, otherwise it will generate a random password.

If the specified user is from a parent Domain, the installer will use that user.
If the user doesn't exist, it will create the user in the child domain (the domain that the controller is joined to).

> The installer will never create a user in the parent Domain.

Reviewer 2: ```suggestion
The installer will never create a user in the parent domain.
```

##### Read-Only Domain Controllers

> The installer can only use an existing domain account when installing on a Read-Only Domain Controller.

Reviewer 2: ```suggestion
The installer can use only an existing domain account when installing on a read-only domain controller.
```

## Upgrade

> When upgrading the Datadog Agent on a domain controller or host where the user has supplied a username for the Agent, it's only required to supply the `DDAGENTUSER_NAME` but not the `DDAGENTUSER_PASSWORD`.

Reviewer 2: ```suggestion
When you upgrade the Datadog Agent on a domain controller or host where the user has supplied a username for the Agent, you must supply the `DDAGENTUSER_NAME` but not the `DDAGENTUSER_PASSWORD`.
```

> ### Installation with Chef

Reviewer 2: This heading level makes it look like this section is a subsection of "Upgrade". Is that accurate, or should this be an H2 `##`?

If no user account is specified on the command line, the installer will create a *domain* account named `ddagentuser` in the controller's domain.

If a user account is specified on the command line, but this user account is not found in the domain, the installer will attempt to create it. If a password was specified, the installer will use that password. Otherwise it will generate a random password.

If the specified user account is from a parent domain, the installer will use that user account.
If the user account doesn't exist, it will create the user account in the child domain (the domain that the controller is joined to).
The installer will never create a user account in the parent domain.

##### Read-only domain controllers

The installer can use only an existing domain account when installing on a read-only domain controller.

### Installation with Chef

If you use Chef and the official `datadog` cookbook to deploy the Agent on Windows hosts, **use version 2.18.0 or above** of the cookbook to ensure that the Agent’s configuration files have the correct permissions

## Upgrade

> When you upgrade the Datadog Agent on a domain controller or host where the user has supplied a username for the Agent, you must supply the `DDAGENTUSER_NAME` but not the `DDAGENTUSER_PASSWORD`.

Reviewer: ```suggestion
On Agent version 7.24 and older, if `DDAGENTUSER_NAME` was supplied during the installation, it must also be provided for upgrades.
```
